### PR TITLE
TG-76: List Timesheet Page

### PR DESCRIPTION
**Pull request by fernandocarbone1**

### CODE REVIEW

File: src/Components/TimeSheets/List/list.module.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    margin-bottom: 10px;
+}
+.row {
+    background-color: rgb(243, 240, 240);
+    border-color: rgb(98, 98, 98);
+    border-style: groove;
+    border-width: 3px;
+    border-radius: 10px;
+    padding: 10px;
+
```
Comment (FefeTroanes) [inline src/Components/TimeSheets/List/list.module.css:20]: Se te escapo una linea vacia

File: src/Components/TimeSheets/index.jsx [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import styles from './time-sheets.module.css';
+import List from './List';
 
 function TimeSheets() {
+  const url = `http://localhost:8080/time-sheets`;
+  console.log(url);
```
Comment (FefeTroanes) [inline src/Components/TimeSheets/index.jsx:6]: Este `console.log` lo dejaste a proposito fer? Porque si no creo que podria eliminarse

**Review by GinaSchP — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     "@testing-library/user-event": "^12.8.3",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
+    "react-icons": "^4.3.1",
```
[the remark this answers is not in the document]
Comment (fernandocarbone1) [reply package.json:11]: Listo, Fede. desinstalado desde Bash.

File: src/Components/TimeSheets/List/list.module.css (hunk 1 above)
Comment (fernandocarbone1) [reply src/Components/TimeSheets/List/list.module.css:20]: Línea borrada.

File: src/Components/TimeSheets/index.jsx (hunk 2 above)
Comment (fernandocarbone1) [reply src/Components/TimeSheets/index.jsx:6]: Lo puse para verlo en consola a modo de prueba. Ya fue eliminado, Fede.

File: package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "@testing-library/user-event": "^12.8.3",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
```
[the remark this answers is not in the document]
Comment (fernandocarbone1) [reply package.json:11]: Lo tenía, Nico. Al menos, siempre estuvo presente. Solo que tenía instalada la librería de react-icons en la línea 11 (ya fue desinstalada), entiendo que estaba en la 12 la de scripts.

**Comment by Clarosluciano** [comment]
## Testing Results

Tested locally on: https://alfonso-trackgenix-app-git-tg-76-list-timeshe-5a11d0-basp-m2022.vercel.app

Flagged checklists in Trello Ticket aren't reproducible from the tested environment. 

Test Results: Failed 

## Bugs found:

Scenario 1:
1. **Given** the user clicks on "Timesheets" button
image

2. **When** clicks on "Add" button
image

3. **Then** the user is redirect to https://alfonso-trackgenix-server.vercel.app/time-sheets/. The API is returning all timesheets
image

Scenario 2:
1. **Given** the user clicks on "Timesheets" button
image

2. **When** clicks on any "Edit" button
image

3. **Then** the user is redirect to https://alfonso-trackgenix-server.vercel.app/time-sheets/null. The API is returning a error
image

Note: I cloned both repositories and merged the content of both branches to do this test.

**Comment by Clarosluciano** [comment]
## Testing Results
Tested locally on: https://alfonso-trackgenix-app-git-tg-76-list-timeshe-5a11d0-basp-m2022.vercel.app/

Test Results: Failed

## Bugs found:

1. Alert don't show the incorrect data or empty fields
image
**Expected behavior:** Alert shows the incorrect data or empty fields

2. Date input is confusing. The user can't determine which format to use. Any combination of date trigger the alert error
image
image
image
image
**Expected behavior:** Date input specifies which format to use or Date input has a date picker

3. a) In "Employee ID" input the id shows like a [Object object]
    b) In "Date" input the date has some numbers of unknown origin (they seem to be from the creation time of the timesheet)
image
**Expected behavior:** The id show a valid id number. The date respect a valid format

4. Can't modify any timesheet, the error aler is triggered and the data of current timesheet disappear
image
image
image
**Expected behavior:** The alert shows where the error is

5. (Not tested yet, I don't want to remove something from the seed if I can't add another). To the left of the any "Edit" button there is a cross button that is used to delete timesheets
image
**Expected behavior:** The cross buttons are more visible and the mouse pointer detects them

**Comment by Clarosluciano** [comment]
## Testing Results
Tested locally on: https://alfonso-trackgenix-app-git-tg-76-list-timeshe-5a11d0-basp-m2022.vercel.app/

Test Results: Failed

## Bugs found:

1. When clicks on the first timesheet for edit the user is redirect to https://alfonso-trackgenix-4dinunj9f-basp-m2022.vercel.app/time-sheets-edit?id=629651bd0842eccc55620b89. The page is blank
image
**Expected behavior:** The page shows the timesheet correctly 
 
2. Date input still has the timesheet creation time and is still confusing to the user
image
**Expected behavior:** Date input specifies which format to use or Date input has a date picker

3. Approved checkbox not works correctly. When the alert with the sucess timesheet edit appears and the user comeback at the timesheets list page, the change wasn't made
image
image
image
**Expected behavior:**  If Approved checkbox is clicked and the timesheet edit is success, then the change must be reflected in the timesheet list

4. Still can't add a new timesheet. If possible, provide valid data to be able to test the rest of the functionalities

**Comment by Clarosluciano** [comment]
## Testing Results
Tested locally on: https://alfonso-trackgenix-app-git-tg-76-list-timeshe-5a11d0-basp-m2022.vercel.app/

Test Results: **Pass**
CRUD operations: **Pass**

**Note:** (Suggestion) the "Task" label should be "Task ID" to avoid user confusion

**Pull request merged**